test/common/router/config_impl_test.cc

```diff
@@ -5410,6 +5441,20 @@ name: foo
   EXPECT_EQ(1, route3->tracingConfig()->getRandomSampling().denominator());
   EXPECT_EQ(3, route3->tracingConfig()->getOverallSampling().numerator());
   EXPECT_EQ(0, route3->tracingConfig()->getOverallSampling().denominator());
+
+  const Tracing::CustomTagMap& custom_tags = route3->tracingConfig()->getCustomTags();
+  std::vector<std::string> custom_tag_views;
+  for (const auto& it : custom_tags) {
+    custom_tag_views.emplace_back(
+        dynamic_cast<const Tracing::GeneralCustomTag*>(it.second.get())->toString());
+  }
+  EXPECT_THAT(custom_tag_views,
+              UnorderedElementsAre("LITERAL|ltag|lvalue", "ENVIRONMENT|etag|E_TAG|",
```

Comment:
Shouldn't the env custom tag have the `e_val` value as set on line 5419 above?

Reply:
Just not present in the string, i'll add it.